msal_extensions/token_cache.py

```diff
@@ -0,0 +1,193 @@
+"""Generic functions and types for working with a TokenCache that is not platform specific."""
+import os
+import sys
+import time
+import errno
+import msal
+from .cache_lock import CrossPlatLock
+
+if sys.platform.startswith('win'):
+    from .windows import WindowsDataProtectionAgent
+elif sys.platform.startswith('darwin'):
+    from .osx import Keychain
+
+
+def get_protected_token_cache(
```

Comment:
What if we skip this helper completely, and just define the following into the `__init__.py`? It would be conceptually one less layer of helper, from the perspective of app devs. And it is [believed to be](https://stackoverflow.com/a/791107) the [pattern from the python os package](https://github.com/python/cpython/blob/v3.8.0b1/Lib/os.py#L48-L89).

```python
if sys.platform.startswith('win'):
    from .token_cache import WindowsTokenCache as TokenCache
elif sys.platform.startswith('darwin'):
    from .token_cache import OSXTokenCache as TokenCache
else:
    from .token_cache import UnencryptedTokenCache as TokenCache
```

And then the`UnencryptedTokenCache` (formerly the  `FileTokenCache`) constructor can have that "no encryption" warning behavior built-in, even without a dedicated flag for it.

```python
class UnencryptedTokenCache(...):
    def __init__(self, ...):
        ...
        warnings.warn("You are using an unprotected token cache for platform {}".format(sys.platform)
        # Note: since this becomes the default behavior now,
        # we do not want to raise RuntimeError(...) which would break an app on a Linux platform
        ...
```

Reply:
I like this :) Something I wouldn't have thought of, but it definitely simplifies the consumption of this API. I'll have those changes in shortly

Reply:
Fixed in a68ba24616da7d9d810a5bc33b2d3d0ff3817244